src/serial.rs

```diff
@@ -230,204 +280,271 @@ macro_rules! hal {
                     apb: &mut <$USARTX as RccBus>::Bus,
                 ) -> Self
                 where
-                    PINS: Pins<$USARTX>,
+                    REMAP: Remap<Periph = $USARTX>,
+                    PINS: Pins<REMAP, POS>,
                 {
-                    // enable and reset $USARTX
-                    $USARTX::enable(apb);
-                    $USARTX::reset(apb);
-
                     #[allow(unused_unsafe)]
                     mapr.modify_mapr(|_, w| unsafe{
-                            w.$usartX_remap().$bit(($closure)(PINS::REMAP))
+                            w.$usartX_remap().$bit(($closure)(REMAP::REMAP))
```

Comment:
This threw me off quite a bit. Suggested comment:

```
// The USARTs have different amounts of remap options which may use 1 or 2 bits in the configuration register. The $bit variable selects wether the size is 1 or 2, and the closure is a closure which returns a bool if 1 bit and a u8 otherwise
```